Boundary traction spherical

mfmweerdesteijn opened the pull request:
Pull Request Checklist. Please read and check each box with an X. Delete any part not applicable. Ask on the [forum](https://community.geodynamics.org/c/aspect) if you need help with any step.

I changed boundary traction function such that it also works when you choose a spherical coordinate system and define the boundary traction in spherical coordinates. Previously, when defining in spherical coordinates, they weren't converted to cartesian, and the applied boundary traction was not the same as defined. I used the same procedure for this as in the boundary velocity function code.

### Before your first pull request:

* [ ] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [ ] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [ ] I have tested my new feature locally to ensure it is correct.
* [ ] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [ ] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

gassmoeller commented on include/aspect/boundary_traction/function.h:
```diff
@@ -98,6 +98,12 @@ namespace aspect
         Utilities::Coordinates::CoordinateSystem coordinate_system;
+
+        /**
+         * Whether to specify velocity in x, y, z components, or
```
```suggestion
         * Whether to specify traction in x, y, z components, or
```

gassmoeller commented on source/boundary_traction/function.cc:
```diff
@@ -83,6 +85,12 @@ namespace aspect
                              "be the depth of the point.");
+          prm.declare_entry ("Use spherical unit vectors", "false",
+                             Patterns::Bool (),
+                             "Specify velocity as $r$, $\\phi$, and $\\theta$ components "
```
```suggestion
                             "Specify traction as $r$, $\\phi$, and $\\theta$ components "
```

gassmoeller reviewed (commented):
Except for the two small typos this looks correct to me. Did you figure out why nothing changes in your model with this PR? Can you also add a test case?

mfmweerdesteijn commented:
Have not figured out yet why test results are the same for before and after these changes using the spherical coordinate system option. Have to leave now, back on it tomorrow morning.

mfmweerdesteijn commented:
@gassmoeller never mind, it does work! I just forgot to set the "Use spherical unit vectors" to true.. I have a 2D chunk test case. Should I do the test for the cartesian option and spherical option and show that results are equal? And now it's a linearly increasing load from 0 to 100 yrs in 2.5 year steps, should I cut it off after 2 time steps for the test?

gassmoeller commented:
It looks like you figured out the indentation issues, and the tests look good, you just need to update the results with the official tester results. Let us know in case you need help with that.

gassmoeller reviewed (approved):
Good to go if the testers succeed.

gassmoeller commented:
The files for the screen output of your tests need to be named `screen-output` like for the other tests not `screen_output`.

gassmoeller commented:
Now you need to update the test results with the results of the reference tester. That is explained in the last paragraph of this page: https://aspect-documentation.readthedocs.io/en/latest/user/extending/testing/writing-tests.html

mfmweerdesteijn commented:
Okay I updated the test results locally. But I cant push it, because I deleted the two files "screen_output" on github and not locally. Now it says I first need to pull from my github branch to the local one, but Im not sure how without making things worse :)

tjhei commented:
@mfmweerdesteijn This is the same thing I helped you with a few days ago:
1. Go to the terminal of the machine where you last worked (if not, you will need to pull first)
2. use ``git apply`` with the diff downloaded from the tester
3. commit and push

If you are stuck, ask someone to walk you through it.

gassmoeller reviewed (approved):
:+1:
